src/cata_tiles.cpp

```diff
@@ -680,43 +716,72 @@ tile_type &cata_tiles::load_tile(JsonObject &entry, const std::string &id, int o
 {
     tile_type &curr_subtile = tile_ids[id];
 
-    if ( entry.has_array("fg") ) {
-        JsonArray fg_array = entry.get_array("fg");
-        while (fg_array.has_more()) {
-            const int fg = fg_array.next_int();
-            if ( fg >= 0 ) {
-                curr_subtile.fg.push_back(fg);
+    // loop through fg and bg
+    for (auto g : {std::make_pair("fg",&curr_subtile.fg), std::make_pair("bg",&curr_subtile.bg)}) {
```

Comment:
Instead of a loop (which is clever, but confusing), put the body of the loop into a separate function and call it once for `fg` and once for `bg`.